text/3550-new-range.md

```diff
@@ -390,7 +474,62 @@ We believe that it is best to keep the `Range` naming for several reasons:
+But the legacy range types have a much larger API surface than other `Iterator`s in the standard library, which typically only implement the various iterator traits and maybe have a `remainder` method. Specifically, there are no iterator types in the standard library which have public fields. Nor do any implement `PartialEq`, `Eq`, `Hash`, `Index`, or `IndexMut`.
+
+`RangeInclusive` especially must take care with equality, hashing, and indexing because it can be exhausted. By removing those impls from the iterator for it, we can prevent that misuse entirely.
+
+One of the strongest arguments for new types is the incorrect `ExactSizeIterator` implementations for `Range<u32 | i32>` and `RangeInclusive<u16 | i16>`. These can be excluded if new iterator types are introduced.
+
+Finally, the cost of adding these iterator types is extremely low, given we're already adding a set of new types for the ranges themselves.
+
+## Inherent `map` should map the bounds, not return an iterator
+
+Some argue that inherent `map` should not return an iterator. Some say that they may expect it to map each bound individually (`(1..11).map(|x| x*2)` -> `2..22`). Others say these methods should return `IntoIterator` types instead.
+
+However, making them return an iterator has many benefits:
+- Matches existing behavior
+- Reduces code churn
+- Act as an entry point for other iterator methods
+
+Adding these convenience methods is unlikely to cause confusion because of how common this pattern already is (if anything, the opposite is true). Plus, it's pretty easy to tell based on the function signature what is going on, and it's simple to document.
+
+Changing the meaning of `(1..11).map(...)` is a huge hazard. There is a lot of existing code, documentation, etc that uses it in the `Iterator` sense. It would be incredibly confusing, especially to a newcomer, to have it do something totally different between editions. Especially since in many cases it could silently change meaning:
+
+```rust
+// Edition 2021
+for n in (1..11).map(|n| n*2) {
+    // n = 2, 4, 6, ...., 16, 18, 20
+}
+// Edition 2024?
+for n in (1..11).map(|n| n*2) {
+    // n = 2, 3, 4, 5, 6, 7, ...., 15, 16, 17, 18, 19, 20, 21
+}
+```
+
+If there is demand for a method that maps the bounds, it should be added under a different name, such as `map_bounds` , perhaps even as a method on `RangeBounds`.
+
+## Implicit conversions (coercions)
+
+This proposal specifically avoids involving any form of implicit conversion. Adding coercions from the new to legacy types would have a few benefits:
+
+- Avoid explicit conversions when migrating automatically to Edition 2024
+- Few (if any) library changes needed to support the new types
```

Comment:
If library means std here then say so.  It's being discussed how ecosystem (libraries) are effected.

Reply:
"library changes" here means "changes to libraries" and refers to the ecosystem.

Reply:
"This proposal" refers to the proposal for implicit conversions, not the RFC as a whole.

Reply:
> It's unclear what explicit means here. Aren't many new `.iter()` calls required by this?

Examples of explicit conversions:
- `.into_iter()`
- `.to_legacy()`
- `.into()`
- `legacy::Range::from(range)`